### PR TITLE
Fixes #30712 - pin shoulda_matcher < 4.4

### PR DESCRIPTION
**Pull request by jlsherrill**
see https://github.com/thoughtbot/shoulda-matchers/issues/1332 for details

### CODE REVIEW

File: bundler.d/test.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
   gem 'rubocop-rails', '~> 2.4.2'
   gem 'poltergeist', '>= 1.18.0', :require => false
   gem 'selenium-webdriver', :require => false
-  gem 'shoulda-matchers', '~> 4.0'
+  gem 'shoulda-matchers', '~> 4.0', '< 4.4'
```
Comment (ekohl) [inline bundler.d/test.rb]: The script that converts this to RPM deps creates something ugly from this. It creates `< 5.0` and `< 4.4`.
```suggestion
  gem 'shoulda-matchers', '>= 4.0', '< 4.4'
```

**Review by ekohl — DISMISSED** [review]

File: bundler.d/test.rb (hunk 1 above)
Comment (jlsherrill) [reply bundler.d/test.rb]: didn't think we created rpms for test gems, but no reason to not doing it the better way for copy/paste reasons :)

File: bundler.d/test.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@
   gem 'rubocop-rails', '~> 2.4.2'
   gem 'poltergeist', '>= 1.18.0', :require => false
   gem 'selenium-webdriver', :require => false
-  gem 'shoulda-matchers', '~> 4.0'
+  gem 'shoulda-matchers', '>= 4.0', '< 4.4'
```
Comment (ezr-ondrej) [inline bundler.d/test.rb:21]: Is it reasonable to expect they are going to fix it in the next release and just exclude the bad release?
```suggestion
  gem 'shoulda-matchers', '~> 4.0', '!= 4.4.0'
```

File: bundler.d/test.rb (hunk 1 above)
Comment (ekohl) [reply bundler.d/test.rb]: Ah, we indeed skip it for test gems so it probably wouldn't have mattered.

File: bundler.d/test.rb (hunk 2 above)
Comment (ekohl) [reply bundler.d/test.rb:21]: From the linked commit, it suggests it started to depend on the Zeitwerk loader. https://github.com/theforeman/foreman/pull/7843 did start the effort to move Foreman to use it, but that's still a draft.

**Review by ezr-ondrej — APPROVED** [review]
Eh, let's fix tests for now :shrug:

**Comment by ezr-ondrej** [comment]
@ekohl could you approve please?

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
This probably needs chery picks.

File: bundler.d/test.rb (hunk 2 above)
Comment (jlsherrill) [reply bundler.d/test.rb:21]: reading the issue: https://github.com/thoughtbot/shoulda-matchers/issues/1332  its unclear at this point.  The change that broke it almost seems like it was intentional, but i don't know that a developer has commented on the issue.

**Comment by ezr-ondrej** [comment]
> This probably needs chery picks.

Only for tests to be passing though.

**Comment by tbrisker** [comment]
2.2 - 2a249b49c
2.1 - 064052fe1

**Comment by ekohl** [comment]
In the issue they mentioned that 4.4.1 should fix it again.

**Pull request merged**